.github/workflows/check_translations.py

```diff
@@ -0,0 +1,59 @@
+import sys
+import os
+import glob
+import yaml
+
+
+def load_yaml_as_dic(file_path):
+    print(f"Opening file: {file_path}")
+    my_dict = {}
+    with open(file_path, 'r', encoding='utf-8') as fp:
+        docs = yaml.safe_load_all(fp)
+        for doc in docs:
+            for key, value in doc.items():
+                my_dict[key] = value
+    return my_dict
+
+
+def recursive_keys(data, keys, prefix):
```

Comment:
The recursive_keys function modifies the keys list in place and also returns it, which might be confusing. You should stick to one method, either modify in place or return a new list.

When concatenating the prefix to the key, you might end up with a key that starts with a slash. A condition to manage the prefix when it is an empty string would be helpful to avoid keys starting with ‘/’.

The recursion doesn’t accumulate the nested keys correctly since the prefix is not preserved at each recursion level.

```python
def load_yaml_as_dic(file_path):
    print(f"Opening file: {file_path}")
    my_dict = {}
    try:
        with open(file_path, 'r', encoding='utf-8') as fp:
            docs = yaml.safe_load_all(fp)
            for doc in docs:
                if isinstance(doc, dict):
                    for key, value in doc.items():
                        my_dict[key] = value
                else:
                    raise ValueError('YAML document is not a dictionary')
        return my_dict
    except yaml.YAMLError as e:
        print(f'Error parsing YAML file: {e}')
        return {}

def recursive_keys(data, keys, prefix=''):
    if isinstance(data, dict):
        for key, value in data.items():
            sub_key = f'{prefix}/{key}' if prefix else key
            if isinstance(value, dict):
                recursive_keys(value, keys, sub_key)
            else:
                keys.append(sub_key)
    return keys
```

Reply:
Good finds Javad. But for this simple script these items are not necessary.